### ISSUE TITLE
Wrong candidate window position when using ibus

### ISSUE THREAD

**Issue by zhmars** [akiyosi/goneovim#119]
When using input method(ibus), the candidate window does not follow the cursor.

Similar issue was reported and fixed in neovim-qt. Could you please check this? 

https://github.com/equalsraf/neovim-qt/issues/490

**Comment by akiyosi** [comment]
Hi :)
Thanks for the report on the issue.

I'm trying to fix this problem by referring to the issue equalsraf/neovim-qt#490.
I'm having a bit of a hard time getting an ibus environment ready in my environment and I'm wondering if it's possible to see a fix for this in your environment?
If possible, please get the binary from the following and try to confirm it.

https://github.com/akiyosi/goneovim/actions/runs/106671549

**Comment by zhmars** [comment]
Thanks. I got the following error when I tried to run it. Please let me know if I'm missing something. 

```conf
LaunchProcess: failed to execvp:
/tmp/goneovim/QtWebEngineProcess
SIGTRAP: trace trap
PC=0x7fd371ad1d91 m=0 sigcode=128
signal arrived during cgo execution

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0xbc27b0, 0xc000159870, 0xc0001598a0)
        runtime/cgocall.go:128 +0x5b fp=0xc000159840 sp=0xc000159808 pc=0x53db1b
github.com/therecipe/qt/webengine._Cfunc_QWebEnginePage_NewQWebEnginePage(0x0, 0x0)
        _cgo_gotypes.go:619 +0x4a fp=0xc000159870 sp=0xc000159840 pc=0xa43c0a
github.com/therecipe/qt/webengine.NewQWebEnginePage.func1(0x0, 0x0, 0x9)
        github.com/therecipe/qt/webengine/webengine-minimal.go:477 +0x66 fp=0xc0001598b0 sp=0xc000159870 pc=0xa5bd96
github.com/therecipe/qt/webengine.NewQWebEnginePage(0x0, 0x0, 0xc0002f59d0)
        github.com/therecipe/qt/webengine/webengine-minimal.go:477 +0x39 fp=0xc0001598f0 sp=0xc0001598b0 pc=0xa4dbe9
github.com/akiyosi/goneovim/editor.newMarkdown(0xc0001e2280, 0xe839a0)
        github.com/akiyosi/goneovim/editor/markdown.go:51 +0x43 fp=0xc000159950 sp=0xc0001598f0 pc=0xaa1e33
github.com/akiyosi/goneovim/editor.newWorkspace(0x0, 0x0, 0x765845, 0xb78540, 0xc000159a50)
        github.com/akiyosi/goneovim/editor/workspace.go:162 +0x611 fp=0xc000159a08 sp=0xc000159950 pc=0xad90a1
github.com/akiyosi/goneovim/editor.(*Editor).initWorkspaces(0xc0001f4c00)
        github.com/akiyosi/goneovim/editor/editor.go:271 +0x421 fp=0xc000159ad0 sp=0xc000159a08 pc=0xa96d71
github.com/akiyosi/goneovim/editor.InitEditor()
        github.com/akiyosi/goneovim/editor/editor.go:214 +0x6ad fp=0xc000159f50 sp=0xc000159ad0 pc=0xa964dd
main.main()
        github.com/akiyosi/goneovim/cmd/goneovim/main.go:14 +0x20 fp=0xc000159f60 sp=0xc000159f50 pc=0xb04e60
runtime.main()
        runtime/proc.go:203 +0x21e fp=0xc000159fe0 sp=0xc000159f60 pc=0x569dbe
runtime.goexit()
        runtime/asm_amd64.s:1357 +0x1 fp=0xc000159fe8 sp=0xc000159fe0 pc=0x594dc1

goroutine 18 [select]:
os/exec.(*Cmd).Start.func2(0xc000302000)
        os/exec/exec.go:448 +0xc4
created by os/exec.(*Cmd).Start
        os/exec/exec.go:447 +0x6d0

goroutine 34 [sync.Cond.Wait]:
runtime.goparkunlock(...)
        runtime/proc.go:310
sync.runtime_notifyListWait(0xc0002a8050, 0x1)
        runtime/sema.go:510 +0xf8
sync.(*Cond).Wait(0xc0002a8040)
        sync/cond.go:56 +0x9d
github.com/neovim/go-client/msgpack/rpc.(*Endpoint).dequeueNotifications(0xc00022dba0, 0x1, 0x1, 0x0)
        github.com/neovim/go-client/msgpack/rpc/endpoint.go:630 +0x5d
github.com/neovim/go-client/msgpack/rpc.(*Endpoint).runNotifications(0xc00022dba0)
        github.com/neovim/go-client/msgpack/rpc/endpoint.go:642 +0x20a
created by github.com/neovim/go-client/msgpack/rpc.(*Endpoint).Serve
        github.com/neovim/go-client/msgpack/rpc/endpoint.go:367 +0xea

goroutine 19 [IO wait]:
internal/poll.runtime_pollWait(0x7fd364ecbda8, 0x72, 0xffffffffffffffff)
        runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc0002bc1f8, 0x72, 0x1001, 0x1000, 0xffffffffffffffff)
        internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
        internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc0002bc1e0, 0xc000313000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        internal/poll/fd_unix.go:169 +0x1cf
os.(*File).read(...)
        os/file_unix.go:259
os.(*File).Read(0xc0002291a0, 0xc000313000, 0x1000, 0x1000, 0xc0002eb560, 0xc0000dfdb0, 0x91fd01)
        os/file.go:116 +0x71
bufio.(*Reader).fill(0xc0002bc300)
        bufio/bufio.go:100 +0x103
bufio.(*Reader).ReadByte(0xc0002bc300, 0x0, 0x0, 0x0)
        bufio/bufio.go:252 +0x39
github.com/neovim/go-client/msgpack.(*Decoder).Unpack(0xc00021e1e0, 0x4, 0x0)
        github.com/neovim/go-client/msgpack/unpack.go:154 +0x55
github.com/neovim/go-client/msgpack/rpc.(*Endpoint).Serve(0xc00022dba0, 0x0, 0x0)
        github.com/neovim/go-client/msgpack/rpc/endpoint.go:370 +0xff
github.com/neovim/go-client/nvim.(*Nvim).Serve(0xc000201500, 0x0, 0x0)
        github.com/neovim/go-client/nvim/nvim.go:53 +0x81
github.com/neovim/go-client/nvim.(*Nvim).startServe.func1(0xc000201500)
        github.com/neovim/go-client/nvim/nvim.go:59 +0x2b
created by github.com/neovim/go-client/nvim.(*Nvim).startServe
        github.com/neovim/go-client/nvim/nvim.go:58 +0x6c

goroutine 20 [semacquire]:
sync.runtime_SemacquireMutex(0xc00020152c, 0x0, 0x1)
        runtime/sema.go:71 +0x47
sync.(*Mutex).lockSlow(0xc000201528)
        sync/mutex.go:138 +0xfc
sync.(*Mutex).Lock(...)
        sync/mutex.go:81
github.com/neovim/go-client/nvim.(*Nvim).Serve(0xc000201500, 0x0, 0x0)
        github.com/neovim/go-client/nvim/nvim.go:51 +0xc3
github.com/akiyosi/goneovim/editor.(*Workspace).startNvim.func3(0xc0001e2280)
        github.com/akiyosi/goneovim/editor/workspace.go:332 +0x36
created by github.com/akiyosi/goneovim/editor.(*Workspace).startNvim
        github.com/akiyosi/goneovim/editor/workspace.go:331 +0x2c8

goroutine 21 [chan receive]:
github.com/neovim/go-client/msgpack/rpc.(*Endpoint).Call(...)
        github.com/neovim/go-client/msgpack/rpc/endpoint.go:231
github.com/neovim/go-client/nvim.(*Nvim).call(0xc000201500, 0xd369c5, 0xf, 0xc1b4a0, 0xc00039a090, 0xc000391ae0, 0x1, 0x1, 0xc0003895c0, 0x3)
        github.com/neovim/go-client/nvim/nvim.go:406 +0xdc
github.com/neovim/go-client/nvim.(*Nvim).Option(...)
        github.com/neovim/go-client/nvim/apiimp.go:1202
github.com/akiyosi/goneovim/editor.(*Workspace).getNvimOptions(0xc0001e2280)
        github.com/akiyosi/goneovim/editor/workspace.go:494 +0x121
github.com/akiyosi/goneovim/editor.(*Workspace).init(0xc0001e2280, 0x0, 0x0)
        github.com/akiyosi/goneovim/editor/workspace.go:355 +0xab
created by github.com/akiyosi/goneovim/editor.(*Workspace).startNvim
        github.com/akiyosi/goneovim/editor/workspace.go:342 +0x307

rax    0x0
rbx    0x317d830
rcx    0x7fd3762bd62d
rdx    0x0
rdi    0x0
rsi    0x0
rbp    0x7ffedf980210
rsp    0x7ffedf980100
r8     0x0
r9     0x7ffedf980128
r10    0x1
r11    0x0
r12    0x7ffedf980150
r13    0x1a5f3
r14    0x7ffedf9802f0
r15    0x7ffedf980110
rip    0x7fd371ad1d91
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
```

**Comment by akiyosi** [comment]
Oops, I'm sorry.
It appears that the library of binaries generated by the Github Actions flow is being generated incomplete.
Please wait a while as I try to fix it.

**Comment by akiyosi** [comment]
I found the cause of the non-starting problem.
The binaries generated by Github Actions do not have execution privileges.
Please grant execution privileges to two binaries as follows.
(I will fix the github actions script later.)

```
$ ls -l
total 19000
drwxrwxr-x  7 akiyosi akiyosi     4096 May 17 04:41 ./
drwxr-xr-x  8 akiyosi akiyosi     4096 May 17 04:41 ../
-rw-rw-r--  1 akiyosi akiyosi 19139712 May 17  2020 goneovim
drwxrwxr-x  2 akiyosi akiyosi     4096 May 17  2020 lib/
drwxrwxr-x 34 akiyosi akiyosi     4096 May 17  2020 plugins/
drwxrwxr-x 26 akiyosi akiyosi     4096 May 17  2020 qml/
-rw-rw-r--  1 akiyosi akiyosi   284080 May 17  2020 QtWebEngineProcess
drwxrwxr-x  2 akiyosi akiyosi     4096 May 17  2020 resources/
drwxrwxr-x  3 akiyosi akiyosi     4096 May 17  2020 translations/
$ 
$ chmod u+x goneovim QtWebEngineProcess 
$ ls -l
total 19000
drwxrwxr-x  7 akiyosi akiyosi     4096 May 17 04:41 ./
drwxr-xr-x  8 akiyosi akiyosi     4096 May 17 04:41 ../
-rwxrw-r--  1 akiyosi akiyosi 19139712 May 17  2020 goneovim*
drwxrwxr-x  2 akiyosi akiyosi     4096 May 17  2020 lib/
drwxrwxr-x 34 akiyosi akiyosi     4096 May 17  2020 plugins/
drwxrwxr-x 26 akiyosi akiyosi     4096 May 17  2020 qml/
-rwxrw-r--  1 akiyosi akiyosi   284080 May 17  2020 QtWebEngineProcess*
drwxrwxr-x  2 akiyosi akiyosi     4096 May 17  2020 resources/
drwxrwxr-x  3 akiyosi akiyosi     4096 May 17  2020 translations/
```

**Comment by zhmars** [comment]
Ok. I can run it now after granting the  execution privileges. But it seems no difference with latest stable relelase.

**Comment by akiyosi** [comment]
Hi
This issue was fixed in #119.
Please check with the latest binary.

https://github.com/akiyosi/goneovim/actions/runs/110497499

**Comment by zhmars** [comment]
> Hi
> This issue was fixed in #119.
> Please check with the latest binary.
> 
> https://github.com/akiyosi/goneovim/actions/runs/110497499

Confirmed. Thank you for continuously improving goneovim :)
